components/suggest/src/db.rs

```diff
@@ -395,15 +395,18 @@ impl<'a> SuggestDao<'a> {
         Ok(suggestions)
     }
 
-    /// Fetches Suggestions of type Amo provider that match the given query
-    pub fn fetch_amo_suggestions(&self, query: &SuggestionQuery) -> Result<Vec<Suggestion>> {
+    /// Query for suggestions using the keyword prefix and provider
+    fn map_prefix_keywords<T>(
```

Comment:
I didn't fold in the pocket sql statement with this one since it slightly varies since it does `GROUP BY` confidence and we don't have a suggestion limit for that query either